shellpython/preprocessor.py

```diff
@@ -90,11 +96,25 @@ def _translate_to_temp_path(path):
     :param path: The path to be translated
     :return: The translated path
     """
+    import platform
+
     absolute_path = os.path.abspath(path)
     relative_path = os.path.relpath(absolute_path, os.path.abspath(os.sep))
+
     # TODO: this will not work in win where root is C:\ and absolute_in_path is on D:\
-    translated_path = os.path.join(tempfile.gettempdir(), 'shellpy', _get_username(), relative_path)
-    return translated_path
+    # TODO: test this
+    if platform.system().lower().startswith('win'):
+        strip_pattern = r'^\w:\\(.*)'
+        relative_path = re.findall(strip_pattern, absolute_path)[0]
+
+    if getFolderPath:
+        relative_folder_path, file_name = os.path.split(relative_path)
+        folder_path = os.path.join(tempfile.gettempdir(), 'shellpy', _get_username())
+        translated_path = os.path.join(folder_path, file_name)
+        return translated_path, folder_path
+    else:
+        translated_path = os.path.join(tempfile.gettempdir(), 'shellpy', _get_username(), file_name)
```

Comment:
Tests do not pass here 
UnboundLocalError: local variable 'file_name' referenced before assignment